src/helpers/signUpValidation.js

```diff
@@ -0,0 +1,38 @@
+export const emailTest = (email, emailAvail) => {
+  return email.includes('@') && !emailAvail
+}
+export const emailValid = (email) => {
+  return email.includes('@')
+}
+export const usernameTest = (username, usernameTaken) => {
+  return username.length >= 6 && !usernameTaken
+}
+export const usernameLength = (username) => {
+  return username.length >= 6
+}
+export const passwordTest = (password) => {
+  const re = /^(?=.*\d)(?=.*[A-Z])(?=.*[@#$!%&*(),.?":{}|<>^+=])(?!.*[^a-zA-Z0-9@#$!%&*(),.?":{}|<>^+=])(.{8,35})$/
+  return re.test(password)
+}
+export const passwordCapital = (password) => {
+  const re = /^(?=.*[A-Z])/
+  return re.test(password)
+}
+export const passwordLower = (password) => {
+  const re = /^(?=.*[a-z])/
+  return re.test(password)
+}
+export const passwordSpecial = (password) => {
+  const re = /^(?=.*[@#$!%&*(),.?":{}|<>^+=])/
+  return re.test(password)
+}
+export const passwordNumber = (password) => {
+  const re = /^(?=.*[0-9])/
+  return re.test(password)
+}
+export const passwordLength = (password) => {
+  return password.length >= 8
+}
+export const passwordConfirmationTest = (pw, pw2) => {
+  return pw === pw2
+}
```

Comment:
can we consolidate this into a single function that will check all inputs and return an object with the relevant errors? For example, if invalid username length but everything else okay, return `{ usernameLength: true }`

Then call this function on every input change (`handleChange`) and update `this.state.errors` with the output

Reply:
Yes. I will try something along those lines.

Reply:
I began dabbling with this issue a little and came across this [link](https://stackoverflow.com/questions/34714826/better-to-export-an-object-containing-function-or-just-export-multiple-function).

It discusses the pros and cons of an object of functions or multiple functions. It doesn't solve this particular issue, but I thought it was interesting. It did lead to this bit of brilliance, though.

**This**
```js
import {
  emailTest,
  emailValid,
  usernameTest,
  usernameLength,
  passwordTest,
  passwordLength,
  passwordCapital,
  passwordLower,
  passwordSpecial,
  passwordNumber,
  passwordConfirmationTest
} from '../../helpers/signUpValidation'
```
**To This**
```js
import * as validations from '../../../helpers/signUpValidation'
```